This patch consists of a single hunk extracted from upstream commit
aba27009f4bdc4e5f60b501a6ca202cace45afd6

diff --git a/src/lib.rs b/src/lib.rs
index 8cb6249c5..1f77b73a4 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -5,5 +4,8 @@
 )]
-#![deny(missing_docs, warnings)]
-// Don't fail on links to things not enabled in features
-#![allow(unknown_lints, intra_doc_link_resolution_failure)]
+#![allow(
+    unknown_lints,
+    renamed_and_removed_lints,
+    intra_doc_link_resolution_failure,
+    broken_itra_doc_links
+)]
 //! Library for easier and safe Unix signal handling
